web/src/components/vis/use/usePlotData.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  computed, Ref, ref, onActivated, onDeactivated, watchEffect,
+} from '@vue/composition-api';
+import store from '../../../store';
+import { LOAD_PLOT } from '../../../store/actions.type';
+import { SET_PLOT } from '../../../store/mutations.type';
+
+export default function useDataPlot(id: Ref<string>, plotName: string) {
+  const active = ref(true);
+  const dataset = computed(() => store.getters.dataset(id.value));
+  const plot = computed(() => store.getters.plot(id.value, plotName));
+
+  // disable-eslint-next-line @typescript-eslint/no-explicit-any
+  function changePlotArgs(args: any) {
+    store.commit(SET_PLOT, {
+      dataset_id: id.value,
+      name: plotName,
+      obj: { args: { ...plot.value.args, ...args } },
+    });
+  }
+
+  onActivated(() => { active.value = true; });
+  onDeactivated(() => { active.value = false; });
+  watchEffect(() => {
+    // TODO I think this is cleaner, but I'm not sure if will register properly.
+    // It needs to watch for changes to plot, active, and id
```

Comment:
In general, code >>> comments.  I'd rather have clearly established dependencies at the top of a watchEffect() than a probably wrong comment about what the watchEffect function used to depend on.

I'm rather fond of writing computed dependencies like this:

```
computedThing() {
  // establish dependencies
  const { foo, bar, baz } = this;
  // then, use of "this" is prohibited by convention.
}
```

This actually has a performance penalty if your logical branching is unbalanced: For example if foo is always false, but inside some foo=true conditional you establish the bar dependency, and bar gets updated 100 times per second, this would be a real problem.   In most cases, though, the performance-vs-readability trade-off is well worth using the above style.

I don't know what the equivalent watchEffect style would be, but in my head it looks something like this:

```suggestion
  const plotVal = plot.value;
  const activeVal = active.value;
  const idVal = id.value;
```

I'd like feedback from others here.

Reply:
I second the practice of pulling out reactive dependencies into local variables in the function "header". I try to do this using destructuring assignment (as @subdavis demoed above) whenever possible. Since that's not possible here, a sequence of `const` variable declarations specifying which refs are used works just as well.

Reply:
I agree with this as well, putting the reactive dependencies at the top of the function makes it much more readable and easier to figure out where/why UI changes occur. As far as syntax for `watchEffect` dependencies, I think the way @subdavis described is the best - at least I can't think of a better solution. 

I do like how clean/self-documenting the destructuring assignment with `this` for computed properties is, I haven't seen that before.

Reply:
In my defense I meant for the comment to be more of an RFC than a permanent feature.
I like the pseudo-destructuring pattern, it feels like the best solution to me (although I named things `plotValue` instead of `plotVal`).

Reply:
> In my defense

I took it to be just a comment, rather than something permanent :smile: 

That's also why I asked for it to be removed before merge (once you were happy with the function on the whole).